typescript_test/test.ts

```diff
@@ -164,8 +164,12 @@ declare function connect<S, P, R>(
   selector: ParametricSelector<S, P, R>
 ): (component: Component<P & R>) => Component<P>
 
+declare function connect2<S, P, R>(
```

Comment:
I couldn't get the existing connect arguments to work.

using ParametricSelector implies there is a parameter, but there is not... and this then disagrees with an implied type of having no parameter... sorry I didn't know what to do here and it might be a reason why a different fix is needed.